### ISSUE TITLE
False negative on duplicate sub in different package

### ISSUE THREAD

**Issue by Tux** [gugod/Perl-Critic-TooMuchCode#10]
If a file contains multiple packages, and any two or more of them contain a sub/method with the same name, this raises a false negative.

Consider this file:
```
$ cat test.pl
#!/pro/bin/perl

use strict;
use warnings;

our $VERSION = "1.00";

package Foo;

sub value {
    return 42;
    } # foo

package Bar;

sub value {
    return 2;
    } # value

package main;

print Foo::value () * Bar::value (), "\n";
```
Will cause this warning:
```
$ perl -wc test.pl
test.pl syntax OK
$ perl test.pl
84
$ perlcritic -1 test.pl
test.pl#10.1:   [1 - TooMuchCode::ProhibitDuplicateSub] Duplicate subroutine definition. Redefined at line: 16, column: 1       :sub value {
Exit 2
$
```

**Comment by gugod** [comment]
I pushed a commit to bypass this problem (gives up when multiple `package` stataments are found) 

Once we figure out how to annotate each sub statement with their scope, we could properly fix such issue.

**Comment by gugod** [comment]
Released 0.15 with that "solution".
